cts/cli/regression.validity.exp

```diff
@@ -473,7 +473,7 @@ element rsc_order: Relax-NG validity error : Element constraints has extra conte
 element rsc_order: Relax-NG validity error : Invalid attribute first-action for element rsc_order
 element rsc_order: Relax-NG validity error : Element constraints has extra content: rsc_order
 =#=#=#= Current cib after: Make resulting CIB invalid, and without validate-with attribute =#=#=#=
-<cib epoch="41" num_updates="0" admin_epoch="0" validate-with="none">
+<cib epoch="11" num_updates="0" admin_epoch="0" validate-with="none">
```

Comment:
If I'd known that the straightforward changes to `update_cib_object()` were going to turn into such an ordeal, I wouldn't have made them. But here we are...

I tested using the first cts-cli validity test that changed: https://github.com/ClusterLabs/pacemaker/pull/3403/commits/8d49c56cc392ad16e7b73dad6bf42fcd0654fd8e#diff-711f8daed23fbf12c53cc6813a6d50307dc79787757201c5105f15877170803aL5

For now I'll assume this change has the same cause. Previously, `xml_create_patchset()` correctly viewed newly created elements within `<configuration>` as a config change. Now, it does not. The reason? `pcmk__xml_copy()` doesn't mark the parents or document dirty. It only calls `pcmk__mark_xml_created()` on the copy.

`pcmk__mark_xml_created()` calls `pcmk__mark_xml_node_dirty()` only if the `pcmk__xf_created` flag is NOT already set. But `pcmk__xf_created` IS already set, because that's part of the `new_private_data()` function that gets called when any XML node is created.
```
pcmk__mark_xml_created:
    if (nodepriv && pcmk__tracking_xml_changes(xml, FALSE)) {
        if (!pcmk_is_set(nodepriv->flags, pcmk__xf_created)) {
            pcmk__set_xml_flags(nodepriv, pcmk__xf_created);
            pcmk__mark_xml_node_dirty(xml);
        }
        ...
    }
```

This has far-reaching implications beyond what changed in this commit. It means that for both `pcmk__xe_create()` and `pcmk__xml_copy()`, if a new node or copy is added to an existing node, the chain of parents and the document do NOT get marked dirty. So if the change is in `<configuration>`, it's not detected as a config change, and the epoch doesn't get bumped.

This probably long-standing behavior looks like a bug.

This should be easy to fix (probably within `pcmk__mark_xml_created()` by moving the `pcmk__mark_xml_node_dirty()` call outside the innermost `if`). I hope it doesn't break other code that was built around the buggy behavior.

Reply:
If that's the case, we can simply drop the `if`

Reply:
The `if` was added in 5125ecbdfe, but it seems unrelated to the stated purpose of the commit, and I don't see any reason for it. I suspect it was a mistake.

Reply:
> drop the `if`

yep that's what I ended up doing :)

> I don't see any reason for it

Maybe
* Avoid logging for setting a bit that's already set -- but `pcmk__set_xml_flags()` uses `LOG_NEVER` so it's a no-op aside from `|=`.
* Avoid wasteful calls to `pcmk__mark_xml_node_dirty()`. Note that for every descendant node, `pcmk__mark_xml_node_dirty()` would trace back through the entire parent chain and doc. But that's avoidable and I've avoided it in my changed commit.